rs/ic_os/config/src/main.rs

```diff
@@ -195,6 +205,57 @@ pub fn main() -> Result<()> {
 
             Ok(())
         }
+        Some(Commands::GenerateGuestosConfig {
+            hostos_config_json_path,
+            guestos_config_json_path,
+            guestos_ipv6_address,
+        }) => {
+            let hostos_config_json_path = Path::new(&hostos_config_json_path);
+
+            let hostos_config: HostOSConfig =
+                serde_json::from_reader(File::open(hostos_config_json_path)?)?;
+
+            // update select file paths for GuestOS
+            let mut guestos_icos_settings = hostos_config.icos_settings;
+            let guestos_config_path = Path::new("/boot/config");
+            if let Some(ref mut path) = guestos_icos_settings.ssh_authorized_keys_path {
+                *path = guestos_config_path.join("accounts_ssh_authorized_keys");
+            }
+
+            let mut guestos_network_settings = hostos_config.network_settings;
+            // Update the GuestOS networking if `guestos_ipv6_address` is provided
+            if let Some(guestos_ipv6_address) = guestos_ipv6_address {
+                match &guestos_network_settings.ipv6_config {
+                    Ipv6Config::Deterministic(deterministic_ipv6_config) => {
+                        guestos_network_settings.ipv6_config = Ipv6Config::Fixed(FixedIpv6Config {
+                            address: guestos_ipv6_address,
+                            gateway: deterministic_ipv6_config.gateway,
+                        });
+                    }
+                    _ => {
+                        anyhow::bail!(
+                            "HostOSConfig Ipv6Config should always be of type Deterministic. Cannot reassign GuestOS networking."
+                        );
+                    }
+                }
+            }
```

Comment:
See how we have to update the GuestOS config because guestOS currently expects to be passed a fixed ipv6 address 😣